herbert_core/utilities/classes/@object_lookup/object_lookup.m

```diff
@@ -89,55 +89,112 @@
             % Input:
             % ------
             %   objects     Object array, or cell array of object arrays
+            %               or cell array of unique_objects_containers
 
 
             if nargin==1 && isstruct(objects)
                 % Assume trying to initialise from a structure array of properties
                 obj = object_lookup.loadobj(objects);
 
             elseif nargin>0
+                
+                if isa(objects, 'unique_objects_container')
+                    
+                    nw = numel(objects); % number of unique-object-containers
+                    
+                    % it is not possible to distinguish access of the first
+                    % element of a scalar unique_objects_container from access of
+                    % the first unique_objects_container in a
+                    % cell of such containers, when using a subscript. So the
+                    % cases are distinguished in this code
+                    if nw == 1
+                        nel = objects.n_runs;
+                        sz  = objects.runs_sz;
+                        obj_all = objects;
+                    else
+                        nel = arrayfun( @(x)get_nruns(x), objects );
+                        sz = arrayfun( @(x)runs_sz(x), objects, 'uniformoutput',false );
```

Comment:
If passing unmodified args like this, don't need to explicitly declare anonymous function, can just use function handle.
```suggestion
                        nel = arrayfun( @get_nruns, objects );
                        sz = arrayfun( @runs_sz, objects, 'uniformoutput',false );
```

Reply:
yes